src/lib/objects/index.d.ts

```diff
@@ -31,22 +31,30 @@ export interface WakuObjectState {
 	readonly profile: User
 	readonly users: User[]
 	readonly tokens: Token[]
+	readonly chatName: string
 }
 
-type StoreType = JSONSerializable
-type DataMessageType = JSONSerializable
-
-export interface WakuObjectContext extends WakuObjectAdapter {
+export interface WakuObjectContext<
+	StoreType extends JSONSerializable = JSONSerializable,
+	DataMessageType extends JSONSerializable = JSONSerializable,
+	ViewType extends string = string,
```

Comment:
What is the reason for making this generic?

Reply:
For a type safety. For example the `onViewChange` function now only allows specific string literals or the store is properly typed. Note that providing types is optional.